StreamDeck.Net/StreamDeckSocketHandler.cs

```diff
@@ -12,43 +12,78 @@
 
 namespace StreamDeck.Net
 {
-    public delegate Task StreamDeckEventHandler(object source, StreamDeckEventPayload e);
+    public delegate Task StreamDeckSocketEventHandler(StreamDeckSocketHandler source, StreamDeckEventPayload e);
+    public delegate Task StreamDeckEventHandler(StreamDeckClient source, StreamDeckEventPayload e);
 
+    /// <summary>
+    /// Wrapper for managing WebSocket communcation with the Stream Deck.
+    /// </summary>
     public class StreamDeckSocketHandler
     {
-        private readonly Uri _uri;
-        private bool _connected;
-        private readonly string _registerEvent;
-        private readonly string _pluginUuid;
-        private readonly JsonSerializer _serializer;
+        /// <summary>
+        /// WebSocket used for communication with Stream Deck
+        /// </summary>
+        [Obsolete("This will likely be made private. It is exposed to allow for custom usage while the library is being built")]
```

Comment:
```suggestion
        [Obsolete("This will likely be made private in the near future. It is exposed to allow for custom usage while the library is being built.")]
```